app/controllers/SubmitVatReturnController.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import javax.inject.Inject
+import models.Error._
+import models.{InvalidJsonResponse, VatReturnDetail}
+import play.api.Logger
+import play.api.libs.json.Json
+import play.api.mvc.{Action, AnyContent}
+import services.VatReturnsService
+import uk.gov.hmrc.play.bootstrap.controller.BaseController
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class SubmitVatReturnController @Inject()(vatReturnsService: VatReturnsService)
+                                         (implicit ec: ExecutionContext) extends BaseController {
+
+  def submitVatReturn(vrn: String): Action[AnyContent] = Action.async { implicit request =>
+    val requestAsJson: Option[VatReturnDetail] = request.body.asJson.flatMap(_.asOpt[VatReturnDetail])
```

Comment:
Hi pal. Could you add a JsonParser to ensure that the returned body is Json. There's an example in Deregister-vat's DataController if you need it